web3.js/src/connection.ts

```diff
@@ -2120,6 +2120,18 @@ export type ConnectionConfig = {
   confirmTransactionInitialTimeout?: number;
 };
 
+function warnListenerRemovedWithoutActiveSubscription(
+  id: number,
+  label: string,
+) {
+  console.warn(
+    `You removed a(n) ${label} subscription with id \`${id}\` ` +
+      'where no active subscription having that id could be . ' +
+      'found. Verify that your code does not try to ' +
+      'unsubscribe the same listener more than once.',
+  );
+}
+
```

Comment:
This commit is from #24607. It shows up here because GitHub doesn't support stacked PRs properly.